adapters/repos/db/shard_async_replication.go

```diff
@@ -293,17 +306,19 @@ func (s *Shard) initAsyncReplication() error {
 		return err
 	}
 
+	err = s.store.PauseCompaction(ctx)
+	if err != nil {
+		return err
+	}
+	defer s.store.ResumeCompaction(ctx)
```

Comment:
what is compaction paused for?
rest of the method is called in goroutine, wouldn't compaction be resumed right after pausing?